pkg/sources/circleci/circleci.go

```diff
@@ -72,30 +78,46 @@ func (s *Source) Init(_ context.Context, name string, jobId, sourceId int64, ver
 func (s *Source) Chunks(ctx context.Context, chunksChan chan *sources.Chunk) error {
 	projects, err := s.projects(ctx)
 	if err != nil {
-		return err
+		return fmt.Errorf("error getting projects: %w", err)
 	}
 
+	var scanned uint64
+	var scanErrs []error
 	for _, proj := range projects {
-		builds, err := s.buildsForProject(ctx, proj)
-		if err != nil {
-			return err
-		}
-
-		for _, bld := range builds {
-			buildSteps, err := s.stepsForBuild(ctx, proj, bld)
+		proj := proj
+		s.jobPool.Go(func() error {
+			builds, err := s.buildsForProject(ctx, proj)
 			if err != nil {
-				return err
+				scanErrs = append(scanErrs, fmt.Errorf("error getting builds for project %s: %w", proj.RepoName, err))
```

Comment:
**Suggestion:** we'll need to make this thread safe with a lock or something. Since we're only counting the errors maybe we could use an atomic counter?

Reply:
Ahh yea, we also log the errors so i'll make a struct to encapsulate all the errors and add a mutex to ensure thread safety.